File: news-flutter/src/main/java/com/angkorteam/news/flutter/widget/FloatingActionButton.java
package com.angkorteam.news.flutter.widget;

import com.angkorteam.news.flutter.Page;
import com.angkorteam.news.flutter.common.Entry;
import com.angkorteam.news.flutter.common.Widget;
import org.jooq.DSLContext;

public class FloatingActionButton extends Widget<FloatingActionButtonAttribute> {

    public FloatingActionButton(Entry<FloatingActionButtonAttribute, Object>... attrs) {
        super(attrs);
    }

    public static Entry<FloatingActionButtonAttribute, Object> child(Widget child) {
        return new Entry<>(FloatingActionButtonAttribute.child, child);
    }

    public static Entry<FloatingActionButtonAttribute, Object> tooltip(String tooltip) {
        return new Entry<>(FloatingActionButtonAttribute.tooltip, tooltip);
    }

    public static Entry<FloatingActionButtonAttribute, Object> onPressed(String onPressed) {
        return new Entry<>(FloatingActionButtonAttribute.onPressed, onPressed);
    }

    @Override
    public void store(DSLContext context, Page page) {
        store(context, page, FloatingActionButtonAttribute.values());
        super.store(context, page);
    }

}

File: news-mobile/lib/function/floating_action_button.dart
import 'package:flutter/material.dart';
import 'package:mobile/clients/news_client.dart';
import 'package:mobile/function/signal.dart';
import 'package:mobile/function/widget.dart';

FloatingActionButton buildFloatingActionButton(
    NewsClient client, BuildContext context, Map<String, dynamic> attrs) {
  var tooltip = attrs['tooltip'] as String;

  Widget child = buildWidget(
    client,
    context,
    attrs['child']['type'],
    attrs['child']['attrs'],
  );

  return FloatingActionButton(
    tooltip: tooltip,
    child: child,
    onPressed: () => {signal(client, context, attrs)},
  );
}
